### PR TITLE
Fix missing dangling comments in for loop

### PR DESCRIPTION
**Pull request by yamafaktory**
This should solve #697.

### CODE REVIEW

File: src/printer.js [hunk 1]
```diff
@@ -987,9 +987,24 @@ function genericPrintNoParens(path, options, print) {
...
       if (!n.init && !n.test && !n.update) {
-        return concat(["for (;;)", body]);
+        return concat([
+          "for ",
+          hasSomeDanglingLineComment
+            ? concat([
+              comments.printDanglingComments(path, options, true),
+              line
+            ])
+            : comments.printDanglingComments(path, options, true),
```
Comment (vjeux) [inline src/printer.js]: What happens if you just put this condition in? The comment is not going to appear at the same place but if it is stable, then this is okay.

**Comment by vjeux** [comment]
Okay, I don't think that we should respect the comment being in-between the for and the parenthesis. If we make it just above the for loop then it makes the implementation easier.

```diff
@@ -970,11 +971,15 @@ function genericPrintNoParens(path, options, print) {
     case "ForStatement": {
       const body = adjustClause(path.call(print, "body"), options);

+      const dangling = comments.printDanglingComments(path, options, /* sameLine */ true);
+      const printedComments = dangling ? concat([dangling, softline]) : "";
+
       if (!n.init && !n.test && !n.update) {
-        return concat(["for (;;)", body]);
+        return concat([printedComments, "for (;;)", body]);
       }

       return concat([
+        printedComments,
         "for (",
         group(
           concat([
```

input:

```js
for // comment
(;;);

for /* comment */(;;);
```

output:

```js
// comment
for (;;);

/* comment */
for (;;);
```

What do you think?

**Comment by yamafaktory** [comment]
@vjeux You're right, this makes sense 👍. Moreover, I have never seen comments positioned like this in any codebase. All done 🚀.

**Comment by jlongster** [comment]
👍 Awesome, thanks!

**Comment by jlongster** [comment]
@yamafaktory I wish I had said this before merging, but do you mind adding a comment to this code explaining exactly what it fixes? We'll look at this later and wonder what it's fixing, so comments are very helpful to track what all these edge cases are. Can you open a new PR with that comment since I already merged this? Sorry.

**Comment by yamafaktory** [comment]
@jlongster Absolutely no problem 👍.

**Pull request merged**